### ISSUE TITLE
M5 - Service Access Information: Uplift to TS 26.512 V17.7.0

### ISSUE THREAD

**Issue by rjb1000** [5G-MAG/rt-5gms-application-function#113]
# Background

The following changes to TS 26.512 were agreed at SA4#126 (Chicago) and approved at SA#102 (Edinburgh):

* CR0054 "Essential maintenance" in [S4-231631](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_126_Chicago/Docs/S4-231631.zip) and plenary CR Pack [SP-231368](https://www.3gpp.org/ftp/TSG_SA/TSG_SA/TSGS_102_Edinburgh_2023-12/Docs/SP-231368.zip).

The updated specification will be published shortly as TS 26.512 V17.7.0.

This issue captures the changes that need to be made to the reference implementation of the 5GMS AF Service Access Information API at reference point M5.

# Summary of changes

1. Due to the change in the data type, the AF now needs to de-reference the reporting interval for the Consumption Reporting Configuration associated with the Provisioning Session.
2. The AF now needs to check if `is_access_reporting` is present in the M1 Consumption Reporting Configuration and should provide the specified default value at M5 if that Boolean is false.
3. The AF now needs to check if `is_access_reporting` is present in the M1 Consumption Reporting Configuration and should provide the specified default value at M5 if that Boolean is false (The Media Session Handler reference implementation needs to be changed as well).

Until the the OpenAPI YAML is updated, the changes are available on the draft branch:

- https://forge.3gpp.org/rep/all/5G_APIs/-/tree/Rel17-draft-TSG102

**Comment by devbbc** [comment]
The AF can now:

- De-reference the reporting interval for the Consumption Reporting Configuration associated with the Provisioning Session.
- Populate client consumption reporting configuration based on the M1 Consumption Reporting Configuration.
- Includes an associated array of bindings containing validPolicyTemplateId and externalReference pairs in the Service Access Information.

Sample output:

```
$ curl -H 'User-Agent: AF' -H 'Content-Type: application/json' -d @consumption-report.json -v http://127.0.0.23:7777/3gpp-m1/v2/provisioning-sessions/3306168c-aa43-41ee-9e7e-315888bca701/consumption-reporting-configuration
> POST /3gpp-m1/v2/provisioning-sessions/3306168c-aa43-41ee-9e7e-315888bca701/consumption-reporting-configuration HTTP/1.1
> Host: 127.0.0.23:7777
> Accept: */*
> User-Agent: AF
> Content-Type: application/json
> Content-Length: 164
>
 
< HTTP/1.1 204 No Content
< Connection: Keep-Alive
< Server: 5GMSAF-af.localdomain/17 (info.title=M1_ConsumptionReportingProvisioning; info.version=2.1.0) rt-5gms-application-function/1.4.0
< Date: Wed, 03 Jan 2024 14:28:55 GMT
<
 
$ curl -H 'User-Agent: AF' -v 'http://127.0.0.24:7777/3gpp-m5/v2/service-access-information/3306168c-aa43-41ee-9e7e-315888bca701'
> GET /3gpp-m5/v2/service-access-information/3306168c-aa43-41ee-9e7e-315888bca701 HTTP/1.1
> Host: 127.0.0.24:7777
> Accept: */*
> User-Agent: AF
>
 
< HTTP/1.1 200 OK
< Connection: Keep-Alive
< Content-Length: 764
< Content-Type: application/json
< Server: 5GMSAF-af.localdomain/17 (info.title=M5_ServiceAccessInformation; info.version=2.4.0) rt-5gms-application-function/1.4.0
< Cache-Control: max-age=60
< Last-Modified: Wed, 03 Jan 2024 14:28:59 GMT
< ETag: 15f76e7d76abbeac67fb766e0ef13089571137090d59cb8b64e8b7357aab3c59
< Date: Wed, 03 Jan 2024 14:28:58 GMT
<
{
    "provisioningSessionId":    "3306168c-aa43-41ee-9e7e-315888bca701",
    "provisioningSessionType":  "DOWNLINK",
    "streamingAccess":  {
    },
    "clientConsumptionReportingConfiguration":  {
        "reportingInterval":    10,
        "serverAddresses":  ["http://127.0.0.24:7777/3gpp-m5/v2/"],
        "locationReporting":    false,
        "accessReporting":  false,
        "samplePercentage": 100
    },
    "dynamicPolicyInvocationConfiguration": {
        "serverAddresses":  ["http://127.0.0.24:7777/3gpp-m5/v2/"],
        "policyTemplateBindings":   [{
                "externalReference":    "policy-external-ref-1",
                "policyTemplateId": "5bc7ea1e-aa43-41ee-9e7e-315888bca701"
            }, {
                "externalReference":    "policy-external-ref-2",
                "policyTemplateId": "79f1a160-aa43-41ee-9e7e-315888bca701"
            }],
        "sdfMethods":   ["5_TUPLE"]
    }
 
}
```

**Comment by davidjwbbc** [comment]
Closed by PR #118
